src/components/forms/FormsContainer.tsx

```diff
@@ -547,6 +546,21 @@ const FormsContainer = () => {
   const [value, setValue]  = React.useState(0);
 
   const handleTabChange = (event: any, newValue: number) => {
+
+    // Need future improvements:
+    // Issue : userBlock component on react-dom-router version 6
+    // This condition checks if you are moving on other tabs except newValue = 3 [Source Tab]
+    // When user changes tab and it has some changes or updates on Source tab, 
+    // it will pop a confirmation alert window, once user confirm it, changes will be discarded 
+    // and user can move in different tab else, user will stay in Source tab
+    if (unsavedChanges && newValue !== 3) {
+      if (window.confirm("WARNING: Leaving this page will discard your changes to the schema. Are you sure you want to leave?")) {
```

Comment:
Check if `useContext` or using Redux (saving `unsavedChanges` in the store) will work when navigating to a page and not a tab. If not, then it probably means more investigation is needed.